alacritty/src/window_context.rs

```diff
@@ -183,9 +194,16 @@ impl WindowContext {
             self.display.pending_update.dirty = true;
         }
 
-        // Live title reload.
-        if !config.window.dynamic_title || old_config.window.title != config.window.title {
-            self.display.window.set_title(&config.window.title);
+        // Update title on config reload according to the following table.
+        //
+        // │cli │ dynamic_title │ current_title == old_config ││ set_title │
+        // │ Y  │       _       │              _              ││     N     │
+        // │ N  │       Y       │              Y              ││     Y     │
+        // │ N  │       Y       │              N              ││     N     │
+        // │ N  │       N       │              _              ││     Y     │
+
+        if !preserve_title && !config.window.dynamic_title {
+            self.display.window.set_title(&config.window.identity.title);
```

Comment:
After looking into this one more time, I think there's no need to reload when `dynamic_title` and current title is the same as configs title. Since we start storing title somewhere(likely on the associated window, and each window will likely has its own title).

Reply:
The table is left as you provided for now, will tune it later on if you insist that we should reload title in second case.

Reply:
> After looking into this one more time, I think there's no need to reload when dynamic_title and current title is the same as configs title.

Why not? Are you saying it's not worth the effort or are you saying that you don't think it would be the ideal behavior?

>  each window will likely has its own title

I'd say this is probably incorrect. There aren't *that* many applications setting custom titles that you're guaranteed to have all your terminals with a dynamic title actually with a dynamic title. Chances are the majority of your one-off terminals are all going to have the same title unless it's been overwritten from the CLI. In that case all of those will remain on the old title while your new one-off terminals will get the new one.

Reply:
Should be resolved now.